src/e2ee.rs

```diff
@@ -131,6 +133,58 @@ impl EncryptHelper {
     }
 }
 
+/// Applies Autocrypt header to Autocrypt peer state and saves it into the database.
+///
+/// If we already know this fingerprint from another contact's peerstate, return that
+/// peerstate in order to make AEAP work, but don't save it into the db yet.
+///
+/// Returns updated peerstate.
+// TODO refactoring: may be better in peerstate.rs?
+pub(crate) async fn get_autocrypt_peerstate(
+    context: &Context,
+    from: &str,
+    autocrypt_header: Option<&Aheader>,
+    message_time: i64,
+) -> Result<Option<Peerstate>> {
+    let mut peerstate;
+
+    // Apply Autocrypt header
+    if let Some(header) = autocrypt_header {
+        // TODO performance...
+        // The "from_nongossiped_fingerprint" part is for AEAP:
+        // If we know this fingerprint from another addr,
+        // we may want to do a transition from this other addr
+        // (and keep its peerstate)
+        peerstate = Peerstate::from_nongossiped_fingerprint_or_addr(
```

Comment:
WRT performance: When I last measured, there was a 3% slowndown in the `receive_emails` benchmark (but I'll measure again with Tokio and so on). This seems acceptable, esp. since the other options seemed very complicated (something like, postponing all AEAP-related stuff to dc_receive_imf, and then only needing to do it if the sender is not in the parent chat - I'm happy to elaborate my half-baked ideas more if someone is interested).

Reply:
When I benchmarked again, there was no slowdown visible anymore, so it should be fine.